pkg/rtc/room.go

```diff
@@ -347,11 +347,13 @@ func (r *Room) Join(participant types.LocalParticipant, requestSource routing.Me
 	pw := r.addParticipantWorkerLocked(participant)
 
 	participant.OnStateChange(func(p types.LocalParticipant, state livekit.ParticipantInfo_State) {
+		ri := r.ToProto()
+		pi := p.ToProto()
 		pw.eventsQueue.Enqueue(func() {
```

Comment:
is there any risk that this could race and the protos could be added to the event queue out of order when handlers are called from separate goroutines?

Reply:
Yes, it could happen. That has been the case always. But, I think the handlers are capable of handling it. Definitely needs a bunch more testing though.